src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs

```diff
@@ -439,6 +440,34 @@ public static Task<(bool, ShellOperationResult)> MoveItemAsync(string[] fileToMo
 				return (await moveTcs.Task, shellOperationResult);
 			});
 		}
+		
+		/*********************************************************************************************
+		  This function generates a unique new name for a file in the specified destination folder.
+		  If the destination folder is not accessible, an InvalidOperationException is thrown.
+		**********************************************************************************************/
+		public static string GetUniqueNewName(ShellFolder destination, string originalName)
+		{
+			if (destination == null)
+			{
+				throw new InvalidOperationException("Destination folder is not accessible.");
+			}
+
+			string newName = originalName;
+			int counter = 2;
+
+			// Iterate through the children of the destination folder and check if the new name already exists.
+			// If it does, append a counter to make it unique.
+			foreach (var existingItem in destination.EnumerateChildren(FolderItemFilter.Storage))
+			{
+				if (string.Equals(newName, existingItem.Name, StringComparison.OrdinalIgnoreCase))
+				{
+					newName = $"{originalName} ({counter})";
```

Comment:
I'd suggest building the newname only when the counter value is calculated. It's will prevent unnecessary string assignation. In the case of a lot of files sharing the same name, it'd reduce the cost by quite a margin.

Reply:
In the end, this foreach can be compacted into a linq request that returns the number of elements with name `originalName` inside the folder.
Might be worth to do the change for the readability gain.